src/Microsoft.Identity.Client/Internal/ClientAssertionCertificateWrapper.cs

```diff
@@ -39,9 +41,22 @@ public ClientAssertionCertificateWrapper(X509Certificate2 certificate)
                     string.Format(CultureInfo.InvariantCulture, MsalErrorMessage.CertificateKeySizeTooSmallTemplate,
                         MinKeySizeInBits));
             }
+
+            if (claimsToSign != null && claimsToSign.Count > 0)
```

Comment:
You need to add this outside the #if / #endif . Also, the fact that no .net core unit tests have failed means that you are missing a test.